File: backend/src/HowAreMyFinances.Api/Models/RecurringExpense.cs
namespace HowAreMyFinances.Api.Models;

public sealed record RecurringExpense(
    Guid Id,
    Guid UserId,
    Guid CategoryId,
    string ItemName,
    decimal Amount,
    string? Vendor,
    string? Comment,
    int DayOfMonth,
    bool IsActive,
    bool IsManual,
    DateTime CreatedAt,
    DateTime UpdatedAt
);

public sealed record RecurringExpenseWithCategory(
    Guid Id,
    Guid UserId,
    Guid CategoryId,
    string ItemName,
    decimal Amount,
    string? Vendor,
    string? Comment,
    int DayOfMonth,
    bool IsActive,
    bool IsManual,
    string CategoryName,
    string? CategoryIcon,
    DateTime CreatedAt,
    DateTime UpdatedAt
);

public sealed record CreateRecurringExpenseRequest(
    string ItemName,
    decimal Amount,
    Guid CategoryId,
    string? Vendor,
    string? Comment,
    int DayOfMonth,
    bool IsManual = false
);

public sealed record UpdateRecurringExpenseRequest(
    string? ItemName,
    decimal? Amount,
    Guid? CategoryId,
    string? Vendor,
    string? Comment,
    int? DayOfMonth,
    bool? IsActive,
    bool? IsManual
);

File: shared/types/recurringExpense.ts
export interface RecurringExpense {
  id: string;
  categoryId: string;
  itemName: string;
  amount: number;
  vendor: string | null;
  comment: string | null;
  dayOfMonth: number;
  isActive: boolean;
  isManual: boolean;
  createdAt: string;
  updatedAt: string;
}

export interface RecurringExpenseWithCategory extends RecurringExpense {
  categoryName: string;
  categoryIcon: string | null;
}

export interface CreateRecurringExpenseRequest {
  itemName: string;
  amount: number;
  categoryId: string;
  vendor?: string;
  comment?: string;
  dayOfMonth: number;
  isManual?: boolean;
}

export interface UpdateRecurringExpenseRequest {
  itemName?: string;
  amount?: number;
  categoryId?: string;
  vendor?: string;
  comment?: string;
  dayOfMonth?: number;
  isActive?: boolean;
  isManual?: boolean;
}
